pynicotine/gtkgui/chatrooms.py

```diff
@@ -256,17 +256,22 @@ def join_room(self, msg):
             return
 
         self.pages[msg.room] = tab = ChatRoom(self, msg.room, msg.users)
+        is_global = (msg.room == core.chatrooms.GLOBAL_ROOM_NAME)
+
+        if is_global:
+            self.prepend_page(tab.container, msg.room, focus_callback=tab.on_focus, close_callback=tab.on_leave_room)
+        else:
+            self.append_page(tab.container, msg.room, focus_callback=tab.on_focus, close_callback=tab.on_leave_room)
 
-        self.append_page(tab.container, msg.room, focus_callback=tab.on_focus, close_callback=tab.on_leave_room)
         tab.set_label(self.get_tab_label_inner(tab.container))
 
-        if msg.room in self.autojoin_rooms:
+        if msg.room in self.autojoin_rooms and not is_global:
```

Comment:
This doesn't seem right. What does this do?

Reply:
It is a workaround for the issue described above in https://github.com/nicotine-plus/nicotine-plus/pull/2527#discussion_r1210739489

Reply:
There's a timing issue where the global room is joined before https://github.com/nicotine-plus/nicotine-plus/blob/2f79b8b7ac1bc41871b5d297893acfbb7776b0a5/pynicotine/gtkgui/chatrooms.py#L364-L371 runs. You can probably get around this by not adding the global room name to self.autojoin_rooms.

Reply:
I really wish the server would send a confirmation message when joining the global room, would make things much easier.

Reply:
> Edit: annoying that GitHub doesn't display new review comments until you refresh the page...

No it's my fault yesterday I Resolved the comment before the review was properly solved, today I unresolved it so you'd see it.